tests/smartclone_test.go

```diff
@@ -0,0 +1,97 @@
+package tests
+
+import (
+	"fmt"
+	"strings"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/onsi/ginkgo/extensions/table"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"kubevirt.io/containerized-data-importer/pkg/controller"
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+
+	cdiv1 "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+)
+
+var _ = Describe("[vendor:cnv-qe@redhat.com][level:component]SmartClone tests", func() {
+
+	var sourcePvc *v1.PersistentVolumeClaim
+
+	fillData := "123456789012345678901234567890123456789012345678901234567890"
+	testFile := utils.DefaultPvcMountPath + "/source.txt"
+	fillCommand := "echo \"" + fillData + "\" >> " + testFile
+	rbdStorageClassName := "csi-rbd"
```

Comment:
We cannot assume that a particular storage class name exists in the environment the test is running in, yes in OUR CI it is true, but someone else can try to run the test in their environment that has a different name for the storage class.

The usual pattern to solve a problem like this is pass the storage class name that has a particular property into the tests as an argument, store it in the framework struct, then use it during the tests. There are several examples of this in framework.go, in particular look at the init function.

I would call it something like 'snapshotSCName' in the framework, that way we know that argument to the test provides a storage class that can do snapshots.

Reply:
Done